plugins/woocommerce/src/Internal/Admin/Agentic/AgenticWebhookPayloadBuilder.php

```diff
@@ -0,0 +1,199 @@
+<?php
+declare(strict_types=1);
+
+namespace Automattic\WooCommerce\Internal\Admin\Agentic;
+
+use Automattic\WooCommerce\Enums\OrderStatus;
+use Automattic\WooCommerce\StoreApi\Formatters\MoneyFormatter;
+use Automattic\WooCommerce\StoreApi\Routes\V1\Agentic\Enums\OrderMetaKey;
+use Automattic\WooCommerce\Internal\Agentic\Enums\Specs\OrderStatus as ACPOrderStatus;
+use Automattic\WooCommerce\Internal\Agentic\Enums\Specs\RefundType;
+use WC_Logger_Interface;
+use WC_Order;
+use WC_Order_Refund;
+
+/**
+ * AgenticWebhookPayloadBuilder class
+ *
+ * Builds webhook payloads for the Agentic Commerce Protocol following
+ * the specification for order lifecycle events.
+ *
+ * @since 10.3.0
+ */
+class AgenticWebhookPayloadBuilder {
+	/**
+	 * Logger instance.
+	 *
+	 * @var WC_Logger_Interface
+	 */
+	private $logger;
+
+	/**
+	 * Money formatter instance.
+	 *
+	 * @var MoneyFormatter
+	 */
+	private $money_formatter;
+
+	/**
+	 * Constructor.
+	 *
+	 * @param WC_Logger_Interface $logger Logger instance.
+	 */
+	public function __construct( WC_Logger_Interface $logger ) {
+		$this->logger          = $logger;
+		$this->money_formatter = new MoneyFormatter();
+	}
```

Comment:
Since this will be resolved via `Containner::get`, this can work. We can also `init` like many other existing classes such as [this one](https://github.com/woocommerce/woocommerce/blob/2645ca8e793d4365c8cd2b85c187f9d5f187dd50/plugins/woocommerce/src/Internal/TransientFiles/TransientFilesEngine.php#L59-L68).

```suggestion
	public function __construct( WC_Logger_Interface $logger, MoneyFormatter $money_formatter ) {
		$this->logger          = $logger;
		$this->money_formatter = $money_formatter;
```

Reply:
Ignored this one as it's a bit tricky with the logger.

Reply:
Again, with the direct call to `wc_get_logger()`, this is not an issue anymore in https://github.com/woocommerce/woocommerce/commit/c08df06b579a7850287501fa6c6c53dafaedf300. The `init()` method still instantiates the formatter, but only because it is so simple.

Reply:
Nice one! :D